### PR TITLE
feat: add timeout for update cluster info

### PR DESCRIPTION
**Pull request by yyzxw**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: controller/clusterinfoupdater.go [hunk 1]
```diff
@@ -20,6 +22,12 @@ import (
 
 const (
 	secretUpdateInterval = 10 * time.Second
```
Comment (ishitasequeira) [inline controller/clusterinfoupdater.go:24]: nit: rename this to `defaultSecretUpdateInterval`

File: controller/clusterinfoupdater.go [hunk 2]
```diff
@@ -20,6 +22,12 @@ import (
...
+
+	EnvClusterInfoTimeout = "ARGO_CD_UPDATE_CLUSTER_INFO_TIMEOUT"
```
Comment (ishitasequeira) [inline controller/clusterinfoupdater.go:26]: We should document this env variable somewhere.

**Review by ishitasequeira — CHANGES_REQUESTED** [review]
@yyzxw Can you add some description around why is this change needed? Do we have an issue or a thread somewhere for this change?

**Comment by yyzxw** [comment]
@ishitasequeira In my actual usage scenario (such as the development environment), there are many clusters that are unreachable to the network, which causes slow updating cluster information

File: docs/operator-manual/high_availability.md [hunk 3]
```diff
@@ -64,6 +64,8 @@ controller replicas. To enable sharding increase the number of replicas in `argo
 and repeat the number of replicas in the `ARGOCD_CONTROLLER_REPLICAS` environment variable. The strategic merge patch below
 demonstrates changes required to configure two controller replicas.
 
+* By default, the controller will update the cluster information every 10 seconds,if there is a problem with your cluster network environment that is causing the update time to take a long time, you can try modifying the environment variables `ARGO_CD_UPDATE_CLUSTER_INFO_TIMEOUT` to increase the timeout time, the unit is seconds.
```
Comment (ishitasequeira) [inline docs/operator-manual/high_availability.md:67]:
```suggestion
* By default, the controller will update the cluster information every 10 seconds. If there is a problem with your cluster network environment that is causing the update time to take a long time, you can try modifying the environment variable `ARGO_CD_UPDATE_CLUSTER_INFO_TIMEOUT` to increase the timeout (the unit is seconds).
```

**Review by ishitasequeira — APPROVED** [review]
Thanks @yyzxw for addressing the comments!! Overall PR looks good!! Just a small suggestion on the docs.

File: docs/operator-manual/high_availability.md (hunk 3 above)
Comment (yyzxw) [reply docs/operator-manual/high_availability.md]: thanks for your review, I have solved it. PLAT

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**